modules/cpmso-yaml-failure-domain-vsphere.adoc

```diff
@@ -0,0 +1,41 @@
+// Module included in the following assemblies:
+//
+// * machine_management/cpmso-configuration.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="cpmso-yaml-failure-domain-vsphere_{context}"]
+= Sample VMware vSphere failure domain configuration
+
+On VMware vSphere infrastructure, the infrastructure resource specification, `infrastructures.config.openshift.io`, defines failure domains for your cluster. The `providerSpec` in the `ControlPlaneMachineSet` custom resource (CR) specifies names for failure domains. A failure domain is an infrastructure resource that comprises a control plane machine set, a vCenter datacenter, vCenter datastore, and a network. 
+
+By using a failure domain resource, you can use a control plane machine set to deploy control plane machines on hardware that is separate from the primary VMware vSphere infrastructure. A control plane machine set also balance control plane machines across defined failure domains to provide fault tolerance capabilities to your infrastructure.
+
+[NOTE]
+====
+If you modify the `ProviderSpec` configuration in the `ControlPlaneMachineSet` CR, the control plane machine set updates all control plane machines deployed on the primary infrastructure and each failure domain infrastructure.
+====
+
+:FeatureName: Defining a failure domain for a control plane machine set
+include::snippets/technology-preview.adoc[]
+
+.Example ProviderSpec configuration with specified failure domain names
+[source,yaml]
+----
```

Comment:
Per the docs guidelines, code blocks for an [object must include the metadata](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#required-fields): 

```
apiVersion: machine.openshift.io/v1
kind: ControlPlaneMachineSet
metadata:
  name: cluster
  namespace: openshift-machine-api
# ...
```

Reply:
Good to know. Updated and I'll be mindful in  future. @rvanderp3, does the metadata YAML lines LGTY with regards to values and indentation?